### PR TITLE
chore: refactoring

### PR DESCRIPTION
**Pull request by ShashwatDadhich**
refactoring of common lib by creating interfaces where necessary

### CODE REVIEW

File: utils/k8s/K8sUtil.go [hunk 1]
```diff
@@ -67,19 +67,75 @@ import (
 )
 
 type K8sUtil struct {
-	logger        *zap.SugaredLogger
+	Logger        *zap.SugaredLogger
 	runTimeConfig *client.RuntimeConfig
 	kubeconfig    *string
 }
 
+type K8sUtilIf interface {
```
Comment (ShashwatDadhich) [inline utils/k8s/K8sUtil.go:75]: Rename It to k8sService

File: utils/k8s/K8sUtil.go [hunk 2]
```diff
@@ -67,19 +67,75 @@ import (
...
+	CheckIfNsExists(namespace string, client *v12.CoreV1Client) (exists bool, err error)
+	CreateNsIfNotExists(namespace string, clusterConfig *ClusterConfig) (err error)
+	GetK8sDiscoveryClientInCluster() (*discovery.DiscoveryClient, error)
+	GetK8sDiscoveryClient(clusterConfig *ClusterConfig) (*discovery.DiscoveryClient, error)
+	GetClientForInCluster() (*v12.CoreV1Client, error)
+	GetCoreV1Client(clusterConfig *ClusterConfig) (*v12.CoreV1Client, error)
+	GetRestConfigByCluster(clusterConfig *ClusterConfig) (*restclient.Config, error)
+}
+
 type ClusterConfig struct {
```
Comment (ShashwatDadhich) [inline utils/k8s/K8sUtil.go:123]: move this to bean.go

File: utils/k8s/K8sUtil.go [hunk 3]
```diff
@@ -67,19 +67,75 @@ import (
 )
 
 type K8sUtil struct {
```
Comment (ShashwatDadhich) [inline utils/k8s/K8sUtil.go:69]: rename it to K8sServiceImpl

File: utils/k8s/K8sUtil.go [hunk 4]
```diff
@@ -67,19 +67,75 @@ import (
...
-	logger        *zap.SugaredLogger
+	Logger        *zap.SugaredLogger
```
Comment (ShashwatDadhich) [inline utils/k8s/K8sUtil.go:70]: logger should be private only

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**